components/prism-core.js

```diff
@@ -235,6 +297,8 @@ var _ = self.Prism = {
 		var rest = grammar.rest;
 
 		if (rest) {
+			_.optimizeGrammar(rest);
```

Comment:
The non-enumerable property `_competition` won't be added to the grammar in the for/in loop that follows.
This seems to raise errors when the code tries to handle competition in `inside` scenarios (like embedded JS in HTML).